databricks/koalas/indexing.py

```diff
@@ -400,16 +399,23 @@ def _select_rows(self, rows_sel):
                 # If slice is None - select everything, so nothing to do
                 return None, None
             elif len(self._internal.index_columns) == 1:
-                start = rows_sel.start
-                stop = rows_sel.stop
-
+                sdf = self._kdf_or_kser._internal.sdf
                 index_column = self._kdf_or_kser.index.to_series()
-                index_data_type = index_column.spark_type
+
+                # get natural order from '__natural_order__' from start to stop
+                # based on index_columns to keep natural order.
+                start = sdf.select(NATURAL_ORDER_COLUMN_NAME) \
+                           .where(index_column._scol == rows_sel.start) \
+                           .first()[0]
```

Comment:
What happens if the given `start` and `stop` don't exist?

Reply:
thanks for the comment

i'll check this with adding more test case related with it.